### PR TITLE
DOC: Ensure pandas warnings & exceptions are always documented in API docs

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

Also adds a pre-commit check that errors in `doc/source/reference/general_utility_functions.rst` are synced with `pandas/errors/__init__.py`

### CODE REVIEW

**Comment by jreback** [comment]
lgtm. @MarcoGorelli if any comments.

File: scripts/pandas_errors_documented.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    pre-commit run pandas-errors-documented --all-files
+"""
+from __future__ import annotations
+
+import argparse
+import ast
+import sys
+from typing import Sequence
+
+API_PATH = "doc/source/reference/general_utility_functions.rst"
```
Comment (MarcoGorelli) [inline scripts/pandas_errors_documented.py:16]: would this work if someone runs this hook on Windows?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Nice! Small comment but the rest looks good

File: scripts/pandas_errors_documented.py (hunk 1 above)
Comment (mroeschke) [reply scripts/pandas_errors_documented.py]: Good call!

**Review by MarcoGorelli — APPROVED** [review]
nice - does this need doing for other parts of the API?

**Comment by mroeschke** [comment]
Yeah I suspect these types of checks could be done elsewhere. It might be trickier for other API sections depending on the import structure.

**Pull request merged**